plasma_framework/contracts/src/exits/payment/PaymentExitGame.sol

```diff
@@ -15,19 +15,21 @@ import "../../utils/OnlyFromAddress.sol";
  */
 contract PaymentExitGame is IExitProcessor, OnlyFromAddress, PaymentStandardExitRouter, PaymentInFlightExitRouter {
     PlasmaFramework private plasmaFramework;
+    bool private initDone = false;
 
     /**
      * @dev use struct PaymentExitGameArgs to avoid stack too deep compilation error.
      */
-    constructor(PaymentExitGameArgs.Args memory args)
-        public
-        PaymentStandardExitRouter(args)
-        PaymentInFlightExitRouter(args)
+    function init(PaymentExitGameArgs.Args memory args) public
     {
-        plasmaFramework = args.framework;
-
+        require(!initDone, "Exit game was already initialized");
+        require(msg.sender == args.framework.getMaintainer(), "Only Maintainer can perform this action");
```

Comment:
True! This is an issue if an attacker knows we're deploying contracts and supersedes our migration script. Is this an issue? The attack window is between:
plasma_framework/migrations/20_deploy_plasma_framework.js
plasma_framework/migrations/140_payment_exit_game.js

Reply:
Looking at how this was done before, it actually had a similar hole. 
You could have deployed another instance of PlasmaFramework and call the constructor of PaymentExitGame before our migration finished.

In my case, the `getMaintainer()` gives an illusion of protection, but 
`require(!initDone, "Exit game was already initialized");`
is actually enough, if we agree that the time window is not a security hole worth patching. Not even sure it's possible.

Reply:
`Looking at how this was done before, it actually had a similar hole.`

In the previous implementation the functionality was in the constructor and there is no way for an attacker to front run as there is not function to call first. If you introduce an init() function then that complicates things and you need to protect it.

Reply:
Why not set the PlasmaFramework in the constructor and the you don't have to derive the address of the PlasmaFramework from the function args.

Reply:
the problem is that if you want a multisig maintainer account, the deployment cannot be streamlined, but cut in half since Vaults need to be activated already.

Reply:
so you'd need to:
1. migrate everything up to Payment Exit Game
2. register vaults
3. continue migration

Reply:
Okay, @boolafish helped me hash this out, what I'll do is:
I'll deploy with PlasmaFramework in the constructor but not init `PaymentStandardExitRouter, PaymentInFlightExitRouter`. I'll do that in the `init` from the maintainer.